pkg/lightning/backend/local/job_worker.go

```diff
@@ -16,20 +16,39 @@ package local
 
 import (
 	"context"
+	"io"
+	"strings"
 	"sync"
+	"time"
 
+	"github.com/pingcap/errors"
 	"github.com/pingcap/failpoint"
+	sst "github.com/pingcap/kvproto/pkg/import_sstpb"
 	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/tidb/br/pkg/logutil"
+	"github.com/pingcap/tidb/pkg/ingestor/ingestcli"
 	"github.com/pingcap/tidb/pkg/lightning/common"
+	"github.com/pingcap/tidb/pkg/lightning/log"
+	"github.com/pingcap/tidb/pkg/lightning/membuf"
 	"github.com/pingcap/tidb/pkg/metrics"
+	"github.com/pingcap/tidb/pkg/util/intest"
+	"github.com/tikv/client-go/v2/oracle"
+	pdhttp "github.com/tikv/pd/client/http"
+	"go.uber.org/zap"
 )
 
+type regionJobWorker interface {
+	run(ctx context.Context) error
+}
+
 type regionJobBaseWorker struct {
 	jobInCh  chan *regionJob
 	jobOutCh chan *regionJob
 	jobWg    *sync.WaitGroup
 
-	doRunJobFn func(ctx context.Context, job *regionJob) error
+	writeFn  func(ctx context.Context, j *regionJob) (*tikvWriteResult, error)
+	ingestFn func(ctx context.Context, j *regionJob) error
+	runJobFn func(ctx context.Context, job *regionJob) error
```

Comment:
My first feeling is write and ingest are sub-steps of runJob, they are not at the same level of concepts. Can you add another precheckFn and delete runJobFn for the special logic of opRegionJobWorker?

Reply:
yes, `writeFn/ingestFn` are part of running one job, they are not part of the worker which runs many jobs, initially I tried to abstract a jobRunner, `op/cloud` have different implementation, and they share the same worker, but it doesn't make code more clearer, so i quit.

> add another precheckFn

good idea, i will try make it a `preRunJobFn`

Reply:
https://github.com/pingcap/tidb/pull/60543/commits/14fcb70432def3eda36184a17cee6ded38d53985